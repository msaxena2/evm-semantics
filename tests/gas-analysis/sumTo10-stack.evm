mkAcct 7

load "account" : { 7 : { "balance" : 1000000000
                       , "nonce"   : 1
                       , "code"    : .Map
                       , "storage" : .Map
                       }
                 }

load "exec" : { "address" : 7
              , "gas"     : 100000
              , "code"    :  PUSH ( 1 , 0 ) ; PUSH ( 1 , 10 ) ; JUMPDEST ;
                             DUP ( 1 ) ; ISZERO ; PUSH ( 1 , 21 ) ; JUMPI ;
                             DUP ( 1 ) ; SWAP ( 2 ) ; ADD ;
                             SWAP ( 1 ) ; PUSH ( 1 , 1 ) ; SWAP ( 1 ) ; SUB ;
                             PUSH ( 1 , 4 ) ; JUMP ;
                             JUMPDEST ; POP ; PUSH ( 1 , 0 ) ; SSTORE ;
                             .OpCodes
              }

start

.EthereumSimulation
