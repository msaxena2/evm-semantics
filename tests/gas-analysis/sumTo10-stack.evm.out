<generatedTop>
  <k> . </k>
  <exit-code> 1 </exit-code>
  <mode> NORMAL </mode>
  <schedule> DEFAULT </schedule>
  <analysis> "blocks" |-> ( ListItem ( { 22 ==> 26 | 20005 | 0 } ) ListItem ( { 21 ==> 21 | 0 | 0 } ) ListItem ( { 10 ==> 20 | 24 | 0 } ) ListItem ( { 5 ==> 9 | 9 | 0 } ) ListItem ( { 0 ==> 4 | 6 | 0 } ) ) </analysis>
  <ethereum>
    <evm>
      <op> . </op>
      <output> .WordStack </output>
      <memoryUsed> 0 </memoryUsed>
      <callDepth> 0 </callDepth>
      <callStack> .List </callStack>
      <interimStates> .List </interimStates>
      <callLog> .Set </callLog>
      <txExecState>
        <program> 0 |-> PUSH ( 1 , 0 ) 2 |-> PUSH ( 1 , 10 ) 4 |-> JUMPDEST 5 |-> DUP ( 1 ) 6 |-> ISZERO 7 |-> PUSH ( 1 , 21 ) 9 |-> JUMPI 10 |-> DUP ( 1 ) 11 |-> SWAP ( 2 ) 12 |-> ADD 13 |-> SWAP ( 1 ) 14 |-> PUSH ( 1 , 1 ) 16 |-> SWAP ( 1 ) 17 |-> SUB 18 |-> PUSH ( 1 , 4 ) 20 |-> JUMP 21 |-> JUMPDEST 22 |-> POP 23 |-> PUSH ( 1 , 0 ) 25 |-> SSTORE </program>
        <id> 7 </id>
        <caller> 0 </caller>
        <callData> .WordStack </callData>
        <callValue> 0 </callValue>
        <wordStack> 21 : ( 10 : ( 0 : .WordStack ) ) </wordStack>
        <localMem> .Map </localMem>
        <pc> 26 </pc>
        <gas> 999979995 </gas>
        <previousGas> 999999995 </previousGas>
      </txExecState>
      <substate>
        <selfDestruct> .Set </selfDestruct>
        <log> .Set </log>
        <refund> 0 </refund>
      </substate>
      <gasPrice> 0 </gasPrice>
      <origin> 0 </origin>
      <gasLimit> 0 </gasLimit>
      <coinbase> 0 </coinbase>
      <timestamp> 0 </timestamp>
      <number> 0 </number>
      <previousHash> 0 </previousHash>
      <difficulty> 0 </difficulty>
    </evm>
    <network>
      <activeAccounts> SetItem ( 7 ) </activeAccounts>
      <accounts>
        <account>
          <acctID> 7 </acctID>
          <balance> 1000000000 </balance>
          <code> .Map </code>
          <storage> 0 |-> 20 </storage>
          <acctMap> "nonce" |-> 1 </acctMap>
        </account>
      </accounts>
      <messages> .MessageCellBag </messages>
    </network>
  </ethereum>
</generatedTop>
